controllers/articles.controller.js

```diff
@@ -1,14 +1,26 @@
-const { fetchArticleById } = require('../models/articles.model');
+const { fetchArticleById, fetchArticles } = require('../models/articles.model');
+
 
 exports.getArticleById = (req, res, next) => {
     const { article_id } = req.params;
-    if (isNaN(article_id)) {
-        return res.status(400).send({ msg: 'Invalid article_id' });
-    }
 
     fetchArticleById(article_id)
         .then((article) => {
             res.status(200).send({ article });
         })
-        .catch(next); 
+        .catch((err) => {
+            if (err.code === '22P02') {  
+                res.status(400).send({ msg: 'Invalid article_id' });
+            } else {
+                next(err); 
+            }
```

Comment:
This isnt how error handling should be dealt with. You should just be passing the error off to next so it can reach your middleware handlers.

Have a look at the notes for this:
https://l2c.northcoders.com/courses/sd-notes/back-end#sectionId=error-handling,step=intro